include/temoto_robot_manager/robot_manager_interface.h

```diff
@@ -104,15 +111,42 @@ class RobotManagerInterface : public temoto_core::BaseSubsystem
     }
   }
 
-  void plan(const geometry_msgs::PoseStamped& pose, std::string planning_group = "")
+  void plan(const std::string& robot_name,
+            const std::string& planning_group,
+            const geometry_msgs::PoseStamped& pose)
   {
     std::string prefix = temoto_core::common::generateLogPrefix(log_subsys_, log_class_, __func__);
     TEMOTO_DEBUG("%s", prefix.c_str());
 
     temoto_robot_manager::RobotPlan msg;
     msg.request.use_default_target = false;
+    msg.request.use_named_target = false;
     msg.request.target_pose = pose;
     msg.request.planning_group = planning_group;
+    msg.request.robot_name = robot_name;
+    
+    if (!client_plan_.call(msg))
+    {
+      throw CREATE_ERROR(temoto_core::error::Code::SERVICE_REQ_FAIL, "Service call returned false.");
+    }
+    else if (msg.response.code == temoto_core::trr::status_codes::FAILED)
+    {
+      throw FORWARD_ERROR(msg.response.error_stack);
+    }
+  }
+
+  void plan(const std::string& robot_name,const std::string& planning_group,const std::string& named_target_pose)
```

Comment:
i guess the name of this method should also be changed to something more descriptive like "planManipulation" or something.